python/private/pypi/parse_requirements.bzl

```diff
@@ -111,8 +112,9 @@ def parse_requirements(
         # The requirement lines might have duplicate names because lines for extras
         # are returned as just the base package name. e.g., `foo[bar]` results
         # in an entry like `("foo", "foo[bar] == 1.0 ...")`.
+        # Lines with different markers are not condidered duplicates.
         requirements_dict = {
-            (normalize_name(entry[0]), _extract_version(entry[1])): entry
+            (normalize_name(entry[0]), _extract_version(entry[1]), requirement(entry[1]).marker): entry
```

Comment:
The same `requirement` function can return you the `version` and the `name` if I remember correctly. We could derive the key for grouping from that instead.